core/src/subgraph/instance.rs

```diff
@@ -0,0 +1,72 @@
+use failure::Error;
+use futures::prelude::*;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use graph::prelude::{SubgraphInstance as SubgraphInstanceTrait, *};
+use graph::web3::types::{Address, Block, Log, Transaction};
+
+pub struct SubgraphInstance<T>
+where
+    T: RuntimeHostBuilder,
+{
+    manifest: SubgraphManifest,
+    hosts: Vec<Arc<T::Host>>,
```

Comment:
I think it's worth documenting here that the ordering of `hosts` matters for correctness, since it's not too obvious unless you think hard about it. Maybe say something about what order they should be in (right now, it matches the ordering of data sources in the manifest file)

Reply:
Done